docs/examples/extending-1-classes/index.md

```diff
@@ -5,200 +5,175 @@ title: Extending Leaflet, Class Theory
-        
-        removeLayer(layer) {
-            …
-            LayerGroup.prototype.removeLayer.call(this, layer);
-        },
-    
-        …
-    });
-
-Calling the parent's constructor is done in a similar way, but using `ParentClass.prototype.initialize.call(this, …)` instead.
+* Including more, or replacing functionality (methods, fields) of an existing class with `L.Class.include()`
+* Using `L.Class.addInitHook()` to run additional constructor code.
 
+## Extending Leaflet Classes
 
-### Naming conventions
+JavaScript is a bit of a weird language. It's not really object-oriented in the traditional sense, but handles inheritance using a [prototype chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Inheritance_and_the_prototype_chain). Only at a later point during its lifetime [syntax was added](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) to allow more 'class-like' inheritance.
 
-When naming classes for Leaflet plugins, please adhere to the following naming conventions:
+Because Leaflet was created before any standardized class syntax existed, it comes with its own implementation for classes in the form of a base class called `L.Class`. Almost all classes in Leaflet extend from `L.Class`, as well as most classes you'll find in plugins. It also contains various utility methods to further modify class behavior.
 
-* Never expose global variables in your plugin.
-* If you inherit one of the existing classes or have a new class, export it and make it available via import.
-* Make your plugin importable like this: `import MyPlugin from 'leaflet-my-plugin'`.
+### Creating a subclass
+
+`L.Class`, or other built-in Leaflet classes derived from it (such as `L.Layer`, `L.Handler`, `L.Control`, etc.), can be extended in the same manner as any other JavaScript class, by using the [`extends`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/extends) keyword. However, unlike regular JavaScript classes, Leaflet classes do not support the [`constructor()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor) method. Instead, constructor logic should go into a special `initialize()` method to preserve backwards compatibility with older versions of Leaflet:
+
+
+```js
+class RotateMarker extends Marker {
+	initialize(latlng, rotation, options) {
+		super.initialize(latlng, options);
+		this._rotation = rotation;
+	}
+}
+```
+
+### Naming conventions
 
+When creating Leaflet classes, adhere to these conventions:
 
+* Class names should be in `UpperCamelCase`.
+* Method and property names should be in `lowerCamelCase`.
+* Private properties and methods start with an underscore (`_`). This indicates they're internal and shouldn't be used directly.
+
+### Setting default options
+
+All classes that extend from `L.Class` can be provided with options, which will automatically be merged when constructed:
+
+```js
+class MyBox extends Class {
+	static {
```

Comment:
This is the first time I see this syntax (executable statements inside a `static` block). I think it'd be a good idea to link to some documentation (MDN?) about this JS feature.